### PR TITLE
Add Blobstores & additional Volumes to Nexus3

### PR DESCRIPTION
**Pull request by Jasper-Ben**
(not yet tested)

### CODE REVIEW

**Comment by stevehipwell** [comment]
@Jasper-Ben I'd prefer it if you used the `extraVolumes` and `extraVolumeMounts` pattern.

```yaml
      volumes:
      {{- with .Values.extraVolumes }}
        {{ tpl . $ | nindent 8 }}
      {{- end }}
```

```yaml
          volumeMounts:
          {{- with .Values.extraVolumeMounts }}
            {{ tpl . $ | nindent 12 }}
          {{- end }}
```

**Comment by stevehipwell** [comment]
@Jasper-Ben the blob store scripting looks good.

File: charts/nexus3/templates/deployment.yaml [hunk 1]
```diff
@@ -109,6 +109,9 @@ spec:
             failureThreshold: 10
             timeoutSeconds: 1
           volumeMounts:
+            {{- with .Values.extraVolumeMounts }}
```
Comment (stevehipwell) [inline charts/nexus3/templates/deployment.yaml:112]: This is tabbed too far in and the extra volume mounts should at the end.

File: charts/nexus3/templates/deployment.yaml [hunk 2]
```diff
@@ -135,6 +138,9 @@ spec:
           resources:
             {{- .Values.resources | toYaml | trim | nindent 12 }}
       volumes:
+        {{- with .Values.extraVolumes }}
```
Comment (stevehipwell) [inline charts/nexus3/templates/deployment.yaml:141]: This is tabbed too far in and the extra volumes should at the end.

File: charts/nexus3/README.md [hunk 3]
```diff
@@ -76,6 +76,8 @@ The following table lists the configurable parameters of the _Nexus3_ chart and
 | `persistence.accessMode`                  | Persistence access mode.                                                                                                         | `ReadWriteOnce`                      |
 | `persistence.storageClass`                | PVC storage class (use `-` for default).                                                                                         | `standard`                           |
 | `persistence.size`                        | Size of PVC to create.                                                                                                           | `8Gi`                                |
+| `extraVolumes`                            | Additional volumes.                                                                                                              | `{}`                                 |
```
Comment (stevehipwell) [inline charts/nexus3/README.md:79]:
```suggestion
| `extraVolumes`                            | Additional volumes.                                                                                                              | `[]`                                 |
```

File: charts/nexus3/README.md [hunk 4]
```diff
@@ -76,6 +76,8 @@ The following table lists the configurable parameters of the _Nexus3_ chart and
...
+| `extraVolumeMounts`                       | Additional mounts.                                                                                                               | `{}`                                 |
```
Comment (stevehipwell) [inline charts/nexus3/README.md:80]:
```suggestion
| `extraVolumeMounts`                       | Additional mounts.                                                                                                               | `[]`                                 |
```

**Comment by stevehipwell** [comment]
@Jasper-Ben you still need to add the extra volume values that you've documented to _values.yaml_.

```yaml
extraVolumes: []
extraVolumeMounts: []
```

**Comment by stevehipwell** [comment]
@Jasper-Ben thank you, now merging!

**Comment by stevehipwell** [comment]
@Jasper-Ben just noticed on the [Nexus docs](https://help.sonatype.com/repomanager3/repository-management/storage-guide) that blob store is two words so I'll update the values accordingly and push back up.

**Comment by Jasper-Ben** [comment]
> @Jasper-Ben I'd prefer it if you used the `extraVolumes` and `extraVolumeMounts` pattern.
> 
> ```yaml
>       volumes:
>       {{- with .Values.extraVolumes }}
>         {{ tpl . $ | nindent 8 }}
>       {{- end }}
> ```
> 
> ```yaml
>           volumeMounts:
>           {{- with .Values.extraVolumeMounts }}
>             {{ tpl . $ | nindent 12 }}
>           {{- end }}
> ```

I believe this is missing a `toYaml`, e.g.:

 ```yaml
           volumeMounts:
           {{- with .Values.extraVolumeMounts }}
             {{ tpl (toYaml . $ | nindent 12) }}
           {{- end }}
 ```

Currently, I get an error when trying to add an extra volume(mount):
`Wait helm template failed. Error: template: nexus3/templates/deployment.yaml:136:19: executing "nexus3/templates/deployment.yaml" at <.>: wrong type for value; expected string; got []interface {}`

**Pull request merged**